galsim/chromatic.py

```diff
@@ -1692,6 +1742,13 @@ def drawImage(self, bandpass, image=None, integrator='trapezoidal', iimult=None,
         Works by finding sums of profiles which include separable portions, which can then be
         integrated before doing any convolutions, which are pushed to the end.
 
+        This method uses a cache to avoid recomputing 'effective' profiles, which are the 
+        wavelength-integrated products of inseparable profiles, the spectral components of 
+        separable profiles, and the bandpass.  Because the cache size is finite, users may find 
+        that it is more efficient when drawing many images to group images using the same 
+        SEDs, bandpasses, and inseparable profiles (generally PSFs) together in order to hit the 
+        cache more often.
```

Comment:
I added a resize method to LRU_Cache and resize methods for each of the specific chromatic caches.  As far as I could tell, this change required either marking the caches with `global` or moving them from the module level to the class level.  I opted for the latter, though this did require me to attach each cache to its class outside of the class definition.